### PR TITLE
[KernelGen][Nvidia] Add softshrink_backward operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `softshrink_backward`. Computes the gradient of the `softshrink` activation. Gradients flow through
only where the magnitude of the original input exceeds `lambd`; elsewhere
the gradient is zero.

## Testing
- Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (15 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Repository / Branch
- Target repository: `flagos-ai/FlagGems`
- Base branch: `master`

## Performance
Test command: `pytest benchmark/test_softshrink_backward.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.501 | 1.657 | 0.906 |
| 64, 64 | 0.006 | 0.029 | 0.189 |
| 4096, 4096 | 0.031 | 0.036 | 0.865 |
| 64, 512, 512 | 0.031 | 0.042 | 0.754 |
| 1024, 1024, 1024 | 1.570 | 1.760 | 0.892 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 3.058 | 3.970 | 0.770 |
| 64, 64 | 0.006 | 0.042 | 0.131 |
| 4096, 4096 | 0.054 | 0.054 | 0.987 |
| 64, 512, 512 | 0.054 | 0.054 | 0.995 |
| 1024, 1024, 1024 | 3.058 | 3.138 | 0.974 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.572 | 1.662 | 0.946 |
| 64, 64 | 0.006 | 0.020 | 0.280 |
| 4096, 4096 | 0.031 | 0.036 | 0.868 |
| 64, 512, 512 | 0.031 | 0.036 | 0.877 |
| 1024, 1024, 1024 | 1.571 | 1.783 | 0.881 |

**Arithmetic Mean Speedup: 0.754x**
## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 0.754 | Primary |
| Tianshu | — | — | — | — |
| Muxi | — | — | — | — |
| Ascend | — | — | — | — |
| Hygon | — | — | — | — |

## Files Changed
- `src/flag_gems/ops/softshrink_backward.py`: Triton kernel implementation
- `tests/test_softshrink_backward.py`: Accuracy test
- `benchmark/test_softshrink_backward.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -14225,6 +14225,22 @@ ops:
...
+  - softshrink_backward
+  labels:
+  - aten
+  - nn.functional
+  - Activation
+  - KernelGen
+  kind:
+  - NeuralNetwork
+  stages:
+  - alpha: '5.4'
```
Comment (bin913) [inline conf/operators.yaml]:
```suggestion
  - alpha: '5.5.0'
```

File: src/flag_gems/ops/softshrink_backward.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@pointwise_dynamic(is_tensor=[True, True, False], promotion_methods=[(0, 1, "DEFAULT")])
+@triton.jit
+def softshrink_backward_kernel(grad_output, self, lambd):
+    # PyTorch semantics: grad flows where |self| > lambd; `not (|self| <= lambd)`
+    # matches aten's comparison predicate including NaN propagation.
+    return tl.where(~(tl.abs(self) <= lambd), grad_output, 0)
+
+
+def softshrink_backward(grad_output, self, lambd):
+    logger.debug("GEMS SOFTSHRINK BACKWARD")
```
Comment (bin913) [inline src/flag_gems/ops/softshrink_backward.py]:
```suggestion
    logger.debug("GEMS SOFTSHRINK_BACKWARD")
```

**Review by bin913 — APPROVED** [review]
